packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -961,9 +959,14 @@ Future<void> _writeWebPluginRegistrant(FlutterProject project, List<Plugin> plug
   final Map<String, Object> context = <String, Object>{
     'methodChannelPlugins': webPlugins,
   };
-  final File pluginFile = project.web.libDirectory.childFile('generated_plugin_registrant.dart');
+
+  assert(project.web.buildDir != null, 'project.web.buildDir must be set before calling _writeWebPluginRegistrant');
+  final File pluginFile = project.web.buildDir!
+    .childFile('web_plugin_registrant.dart');
+
   if (webPlugins.isEmpty) {
     ErrorHandlingFileSystem.deleteIfExists(pluginFile);
+    // Render noop plugin registrant file
```

Comment:
Ah yes, I forgot to implement this, just creating an empty noop plugin registrant file in case there are no plugins, so we don't have to special-case down the line.

Reply:
I've added a noop registerPlugins() {} function so the web_plugins_registrar.dart file can be generated (and used down the line) unconditionally.